test_typing_inspect.py

```diff
@@ -68,6 +69,28 @@ def test_union(self):
         nonsamples = [int, Union[int, int], [], Iterable[Any]]
         self.sample_test(is_union_type, samples, nonsamples)
 
+    def test_optional_type(self):
+        T = TypeVar('T')
+        samples = [type(None),                # none type
+                   Optional[int],             # direct union to none type 1
+                   Optional[T],               # direct union to none type 2
+                   Optional[T][int],          # direct union to none type 3
+                   Union[int, type(None)],    # direct union to none type 4
+                   Union[str, T][type(None)]  # direct union to none type 5
+                   ]
+        nonsamples = [int, Union[int, int], [], Iterable[Any], T, Union[T, str][int]]
+        # unfortunately current definition sets these ones as non samples too (those with nesting only in python <= 3.5)
```

Comment:
Hm, one of the points of `typing_inspect` is to have as much similar behaviour between different versions of Python/`typing` as possible. Is this really hard to fix in 3.5?

Reply:
Actually I was wrong, it seems to pass all of your travis builds. But I suspect that your Travis configuration installs the `typing` backport for all of its tests. Indeed I know that on older (5.3.1 for example) versions of python that include legacy typing modules, it does not work.

Reply:
3.5.1 is notorious for various issues, so I would just not worry about this, in most cases one can just switch to 3.5.2 or 3.5.3, or later.